client/src/App.vue

```diff
@@ -1,5 +1,31 @@
+<script>
+  import modal from '@/components/SettingsModal.vue';
+
+  export default {
+    name: 'app',
+    components: {
+      modal,
+    },
+    data () {
+      return {
+        isModalVisible: false,
+      };
+    },
+    methods: {
+      showModal() {
+        this.isModalVisible = true;
+      },
+      closeModal() {
+        this.isModalVisible = false;
+      }
+    },
+  };
+</script>
+
 <template>
   <v-app id="app">
+    <button type="button" class="btn" @click="showModal">Settings</button>
+    <modal v-show="isModalVisible" @close="closeModal"/>
```

Comment:
We probably don't want to insert the modal in App.vue because then it would be visible even on the login screen. I'm assuming we want it in the Room Menu and Canvas? Then we can just move these to those components